Tests/Common/LoRaDeviceTwin.cs

```diff
@@ -40,6 +39,9 @@ public static Twin Create(LoRaDesiredTwinProperties? desiredProperties = null, L
                 {
                     case null:
                         break;
+                    case LoRaRegionType region:
```

Comment:
Since the code in this branch is the same as in the last `var obj` case, you could remove this and instead change the `Enum` case to the following instead?

```c#
                    case Enum and not LoRaRegionType:
```

Although I suspect it will get to the `IConvertible` case before `var obj`, which is harmless.

Reply:
BTW, I removed this case of `LoRaRegionType` entirely and nothing seems to be breaking?

Reply:
You can collapse the cases to just one pattern:

```diff
diff --git a/Tests/Common/LoRaDeviceTwin.cs b/Tests/Common/LoRaDeviceTwin.cs
index b7853d97..2d692638 100644
--- a/Tests/Common/LoRaDeviceTwin.cs
+++ b/Tests/Common/LoRaDeviceTwin.cs
@@ -39,13 +39,7 @@ namespace LoRaWan.Tests.Common
                 {
                     case null:
                         break;
-                    case LoRaRegionType region:
-                        target[key] = region.ToString();
-                        break;
-                    case uint:
-                    case int:
-                    case Enum:
-                    case bool:
+                    case uint or int or bool or (Enum and not LoRaRegionType):
                         target[key] = value;
                         break;
                     case IConvertible convertible:
```

Reply:
I'll apply your suggestion. It's true that nothing breaks, this is just because we can parse both number and string representations into Enums. However, the real code serializes the region as a string, so to stay close to the real behavior we should serialize it as string as well.

Reply:
I figured that was reasoning, which is why I left in the `Enum and not LoRaRegionType` pattern.

Reply:
Addressed this with 080e6b242d4a21d942b3c6052476229f6246db6c